spec/snippetPreviewSpec.js

```diff
@@ -95,6 +95,57 @@ describe( "The SnippetPreview format functions", function(){
 			expect( snippetPreview.formatCite() ).toBe( "<strong>keyword</strong>/")
 		});
 	});
+
+	describe( "The snippet preview format functions with special characters like periods", function() {
+		// Makes lodash think this is a valid HTML element
+		var mockElement = [];
+		mockElement.nodeType = 1;
+
+		var mockApp = {
+			rawData: {
+				snippetMeta: "This is the Yoast SEO 3.9 release",
+				keyword: "Yoast SEO 3.9"
+			},
+			pluggable: {
+				loaded: true,
+				_applyModifications: function(name, text){return text}
+			}
+		};
+
+		var snippetPreview = new SnippetPreview({
+			analyzerApp: mockApp,
+			targetElement: mockElement
+		});
+
+		it( "should highlight the keywords", function(){
+			expect( snippetPreview.formatMeta() ).toBe( "This is the<strong> Yoast SEO 3.9 </strong>release")
```

Comment:
I totally agree. I've created a new issue for this https://github.com/Yoast/YoastSEO.js/issues/944
The problem isn't in the highlighting, but in the way we capture keywords, so that should be fixed separately to prevent this issue getting huge.